products/PurePhone/services/db/ServiceDB.cpp

```diff
@@ -85,7 +86,7 @@ sys::MessagePointer ServiceDB::DataReceivedHandler(sys::DataMessage *msgl, sys::
 
     case MessageType::DBContactAdd: {
         auto time             = utils::time::Scoped("DBContactAdd");
-        DBContactMessage *msg = reinterpret_cast<DBContactMessage *>(msgl);
+        auto msg              = dynamic_cast<DBContactMessage *>(msgl);
```

Comment:
I believe in that case you can safely use `static_cast` or at least(if you really want to precise), check the result of `dynamic_cast`. This also applies to the rest of code below ;)

Reply:
Changed to `static_cast` :)